File: python/linkedlist/LC_206_ReverseLinkedList.py
# https://leetcode.com/problems/reverse-linked-list/
from typing import Optional
from linkedlist.listnode import ListNode


class Solution:
    def reverseList(self, head: Optional[ListNode]) -> Optional[ListNode]:
        prev = None
        current = head
        while current:
            temp = current.next
            current.next = prev
            prev = current
            current = temp
        return prev

File: java/src/linkedlist/LC_206_ReverseLinkedList.java
package linkedlist;
// https://leetcode.com/problems/reverse-linked-list/

public class LC_206_ReverseLinkedList {

    public ListNode reverseList(ListNode head) {
        ListNode prev = null;
        ListNode current = head;
        while (current != null){
            ListNode temp = current.next;
            current.next = prev;
            prev = current;
            current = temp;
        }
        return prev;
    }

}
